sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_client_async.py

```diff
@@ -215,7 +216,29 @@ async def _do_retryable_operation_async(self, operation, *args, **kwargs):
                         break
                     await asyncio.sleep(self._retry_policy.get_backoff_time(retry_settings, exc))
                     if exc.condition == ErrorCondition.LinkDetachForced:
+                        total_link_credit = self._link.total_link_credit
                         await self._close_link_async()  # if link level error, close and open a new link
+                        if isinstance(self._link, SenderLink):
```

Comment:
are we going to pop the self._link off of the input_handles[] in session.py, or does it reuse the same input handle indices when reattaching?